crates/services/txpool_v2/src/service.rs

```diff
@@ -323,6 +324,29 @@ where
             let mut block_height = self.current_height.write();
             *block_height = new_height;
         }
+
+        // Remove expired transactions
+        let mut removed_txs = vec![];
+        {
+            let mut height_expiration_txs = self.pruner.height_expiration_txs.write();
+            let range_to_remove = height_expiration_txs
+                .range(..=new_height)
+                .map(|(k, _)| *k)
+                .collect::<Vec<_>>();
+            for height in range_to_remove {
+                let expired_txs = height_expiration_txs.remove(&height);
+                if let Some(expired_txs) = expired_txs {
+                    let mut tx_pool = self.pool.write();
+                    removed_txs
+                        .extend(tx_pool.remove_transaction_and_dependents(expired_txs));
+                }
```

Comment:
is it possible that when you call `tx_pool.remove_transaction_and_dependents` you end up removing the same tx twice? i.e once as a dependent for another transaction and once on its own?

Reply:
Yeah or even if a transaction is the dependency of two others. But in the implementation it will not remove twice because of https://github.com/FuelLabs/fuel-core/blob/ccae2d50c4d694a8c77500d2843a3322f694e80b/crates/services/txpool_v2/src/storage/graph.rs#L144

Reply:
okay :)